astropy/io/ascii/core.py

```diff
@@ -1258,6 +1266,10 @@ def __init__(cls, name, bases, dct):
                 connect.io_registry.register_writer(io_format, Table, func)
 
 
+@overload
+def _is_number(x: SupportsFloat) -> Literal[True]: ...
+@overload
+def _is_number(x: object) -> bool: ...
 def _is_number(x):
```

Comment:
```suggestion
def _is_number(x) -> TypeIs[SupportsFloat]:
```

Reply:
But we can already use [TypeGuard](https://docs.python.org/3.11/library/typing.html#typing.TypeGuard).

Reply:
yes, `TypeIs` is new in Python 3.13, but it's also in typing-extensions >= 4.10.0, and feels more appropriate than `TypeGuard` here.

Reply:
> that causes a few test failures 

that's surprising to me, but I guess it makes sense for non Python classes, and in that case I agree that `TypeGuard` is preferable